code/modules/mob/living/simple_animal/hostile/mining/elites/legionnaire.dm

```diff
@@ -0,0 +1,337 @@
+		to_chat(L, "<span class='userdanger'>[src] tramples you and kicks you away!</span>")
+		L.throw_at(throwtarget, 10, 1, src)
+		L.KnockDown(1 SECONDS)
+		L.adjustBruteLoss(melee_damage_upper)
+	addtimer(CALLBACK(src, .proc/legionnaire_charge_2, move_dir, (times_ran + 1)), 0.7)
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/head_detach(target)
+	ranged_cooldown = world.time + 1 SECONDS * revive_multiplier()
+	if(myhead != null)
+		myhead.adjustBruteLoss(600)
+		return
+	if(has_head)
+		has_head = FALSE
+		icon_state = "legionnaire_headless"
+		icon_living = "legionnaire_headless"
+		icon_aggro = "legionnaire_headless"
+		visible_message("<span class='warning'>[src]'s head flies off!")
+		var/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/newhead = new /mob/living/simple_animal/hostile/asteroid/elite/legionnairehead(loc)
+		newhead.GiveTarget(target)
+		newhead.faction = faction.Copy()
+		myhead = newhead
+		myhead.body = src
+		if(health < maxHealth * 0.25)
+			myhead.melee_damage_lower = 40
+			myhead.melee_damage_upper = 40
+		else if(health < maxHealth * 0.5)
+			myhead.melee_damage_lower = 30
+			myhead.melee_damage_upper = 30
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/onHeadDeath()
+	myhead = null
+	addtimer(CALLBACK(src, .proc/regain_head), 50)
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/regain_head()
+	has_head = TRUE
+	if(stat == DEAD)
+		return
+	icon_state = "legionnaire"
+	icon_living = "legionnaire"
+	icon_aggro = "legionnaire"
+	visible_message("<span class='danger'>The top of [src]'s spine leaks a black liquid, forming into a skull!</span>")
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/bonfire_teleport()
+	ranged_cooldown = world.time + 5 * revive_multiplier()
+	if(mypile == null)
+		var/obj/structure/legionnaire_bonfire/newpile = new /obj/structure/legionnaire_bonfire(loc)
+		mypile = newpile
+		mypile.myowner = src
+		playsound(get_turf(src),'sound/items/fultext_deploy.ogg', 200, 1)
+		visible_message("<span class='warning'>[src] summons a bonfire on [get_turf(src)]!</span>")
+		return
+	else
+		var/turf/legionturf = get_turf(src)
+		var/turf/pileturf = get_turf(mypile)
+		if(legionturf == pileturf)
+			mypile.take_damage(100)
+			mypile = null
+			return
+		playsound(pileturf,'sound/items/fultext_deploy.ogg', 200, 1)
+		playsound(legionturf,'sound/items/fultext_deploy.ogg', 200, 1)
+		visible_message("<span class='danger'>[src] melts down into a burning pile of bones!</span>")
+		forceMove(pileturf)
+		visible_message("<span class='danger'>[src] forms from the bonfire!</span>")
+		mypile.forceMove(legionturf)
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/spew_smoke()
+	ranged_cooldown = world.time + 4 SECONDS * revive_multiplier()
+	var/turf/smoke_location = null
+	if(myhead != null)
+		smoke_location = myhead
+	else
+		smoke_location = src
```

Comment:
neither of these are turfs. you want to define `smoke_location` as not a turf

Reply:
was changed